### ISSUE TITLE
Upgrade Codenvy on-prem to 4.6.2 broke workspaces and stacks

### ISSUE THREAD

**Issue by dmytro-ndp** [codenvy/codenvy#644]
Fidelity team encountered blocker problems after updating Codenvy on-prem to version 4.6.2:
1) recent workspaces and projects missing:
![codenvy - recent workspaces and projects missing](https://cloud.githubusercontent.com/assets/1197777/18050558/03b6a6c4-6df8-11e6-8fe4-1df65fac68f5.jpg)

Server were returning *"NullPointerException"* error on getting workspace list;

2) "**ready to go**" stacks list was empty as well as stack library, and it made impossible creating of new workspace:
![codenvy - no ready to go stacks](https://cloud.githubusercontent.com/assets/1197777/18050563/0633828c-6df8-11e6-89dd-d28beaa4e86a.jpg)
![codenvy - no stack library](https://cloud.githubusercontent.com/assets/1197777/18050644/89ea3e9a-6df8-11e6-91e6-e6b2573061e2.jpg)

These issues were caused by improper update of MongoDB collections in time of updating:
- organisation.stacks
- organisation.workspaces2

See additional resources for details:
[report.tar.gz](https://github.com/codenvy/codenvy/files/442346/report.tar.gz)
[mongo.tar.gz](https://github.com/codenvy/codenvy/files/442347/mongo.tar.gz)

[Collection Workspaces2.txt](https://github.com/codenvy/codenvy/files/442469/Workspaces2.txt)
[Collection Stacks.txt](https://github.com/codenvy/codenvy/files/442467/Stacks.txt)

Logs of updating Codenvy on-prem to 4.6.2:
[cli.log.txt](https://github.com/codenvy/codenvy/files/443473/cli.log.txt)
[puppet-agent.log.txt](https://github.com/codenvy/codenvy/files/443474/puppet-agent.log.txt)

Expected migration.log:
> MongoDB shell version: 3.2.1
> connecting to: test
> Workspaces with 'isDev' field: 5
> Workspace configs updated from 'isDev' to 'dev': 5
> Stacks: Stacks with 'isDev' field: 11
> Stacks: Workspace configs updated from 'isDev' to 'dev': 11
> Factory: Factories with 'isDev' field: 0
> Factory: Workspace configs updated from 'isDev' to 'dev': 0

**Comment by mkuznyetsov** [comment]
This issue should be fixed by these scripts (that are scheduled for 4.7.0):
https://github.com/codenvy/codenvy/issues/564
https://github.com/codenvy/codenvy/issues/541

**Comment by bmicklea** [comment]
Will this bug affect all users who have upgraded to 4.6.2 or are there specific triggers?

**Comment by dmytro-ndp** [comment]
It seems migration script didn't update MongoDB properly.
We need additional logs of IM CLI client to check was migration finished properly.

**Comment by bmicklea** [comment]
@dmytro-ndp - here's the CLI log: https://drive.google.com/a/codenvy.com/file/d/0B4Go3CnInfkoNDBicm4tMjFCZ0U/view?usp=sharing

Unfortunately, he wasn't able to find a migration log anywhere on the system even though he knows he executed the migration command from that system.

**Comment by dmytro-ndp** [comment]
@bmicklea: I found *cli.log* which you had sent 22 Aug, and it looks like Santosh was doing something strange to make it possible to update Codenvy on-prem to 4.6.2:
```
2016-08-22 16:01:39 +0000 [SEVERE] com.codenvy.im.console.Console printErrorAndExit {
  "message" : "Artifact '4.6.0' not found",
  "status" : "ERROR"
}  
2016-08-22 16:01:44 +0000 [SEVERE] com.codenvy.im.console.Console printErrorAndExit {
  "message" : "Artifact '4.6' not found",
  "status" : "ERROR"
}  
2016-08-22 16:13:28 +0000 [SEVERE] com.codenvy.im.console.Console printErrorAndExit {
  "artifacts" : [ {
    "artifact" : "codenvy",
    "version" : "4.6.2",
    "status" : "FAILURE"
  } ],
  "message" : "codenvy:4.6.2 is not installable",
  "status" : "ERROR"
}  
2016-08-22 16:14:08 +0000 [SEVERE] com.codenvy.im.console.Console printErrorAndExit {
  "artifacts" : [ {
    "artifact" : "codenvy",
    "version" : "4.6.2",
    "status" : "FAILURE"
  } ],
  "message" : "codenvy:4.6.2 is not installable",
  "status" : "ERROR"
}  
2016-08-22 16:14:36 +0000 [SEVERE] com.codenvy.im.console.Console printErrorAndExit {
  "artifacts" : [ {
    "artifact" : "codenvy",
    "version" : "4.6.2",
    "status" : "FAILURE"
  } ],
  "message" : "codenvy:4.6.2 is not installable",
  "status" : "ERROR"
}  
```
And then the real update to 4.6.2 started:
```
2016-08-22 16:17:15 +0000 [INFO] com.codenvy.im.commands.SimpleCommand execute {'command'='rm -rf /tmp/codenvy; mkdir /tmp/codenvy/; unzip -o /home/centos/codenvy/updates/codenvy/4.6.2/codenvy-4.6.2.zip -d /tmp/codenvy', 'agent'='LocalAgent'}  
2016-08-22 16:17:17 +0000 [INFO] com.codenvy.im.commands.SimpleCommand execute {'command'='if [ ! -f /etc/redhat-release ]; then     exit 1; else     cat /etc/redhat-release | sed 's/.*\s\([0-9\.]*\)\s.*/\1/'; fi', 'agent'='LocalAgent'}
...
```
He commented on this in his mail from 22 Aug:
> I ran into some issues probably because the command is incorrectly stated in the documentation and I have attached my comments regarding the same.

It would very helpful to have answers on the next questions:
1) what exact command did he used to update Codenvy?
2) what version of Codenvy on-prem he had before update?
3) why he tried to update to 4.6.0, 4.6?

Also, zccording to cli.log, file *migration.log* should be somewhere in /**home/centos/** directory.

**Comment by bmicklea** [comment]
Closed - fixed by #564 and #541
